File: src/requests.ts
import * as https from 'https';
import { URLSearchParams } from 'url';
import { getAccessToken } from './access_token';

// Fonction pour effectuer la requête GET playlist_id avec le token
export function searchPlaylistId(token: string, playlistName: string): any {
    const params = new URLSearchParams({
        q: playlistName,
        type: 'playlist',
        limit: '1',
    });

    const options: https.RequestOptions = {
        hostname: 'api.spotify.com',
        path: `/v1/search?${params.toString()}`,
        method: 'GET',
        headers: {
            'Authorization': `Bearer ${token}`,
        },
    };

    return new Promise((resolve, reject) => {
        const req = https.request(options, (res) => {
            let data = '';

            res.on('data', (chunk) => {
                data += chunk;
            });

            res.on('end', () => {
                resolve(JSON.parse(data));
            });
        });

        req.on('error', (error) => {
            reject(error);
        });

        req.end();
    });
}

// Fonction pour effectuer la requête GET playlist avec le token
export function searchPlaylist(token: string, playlistId: string): any {
    const options: https.RequestOptions = {
        hostname: 'api.spotify.com',
        path: `/v1/playlists/${playlistId}`,
        method: 'GET',
        headers: {
            'Authorization': `Bearer ${token}`,
        },
    };

    return new Promise((resolve, reject) => {
        const req = https.request(options, (res) => {
            let data = '';

            res.on('data', (chunk) => {
                data += chunk;
            });

            res.on('end', () => {
                resolve(JSON.parse(data));
            });
        });

        req.on('error', (error) => {
            reject(error);
        });

        req.end();
    });
}

// Fonction pour effectuer la requête GET track avec le token
export async function searchTrack(token: string, trackName: string): Promise<any> {
    const params = new URLSearchParams({
        q: trackName,
        type: 'track',
        limit: '1',
    });

    const options: https.RequestOptions = {
        hostname: 'api.spotify.com',
        path: `/v1/search?${params.toString()}`,
        method: 'GET',
        headers: {
            'Authorization': `Bearer ${token}`,
        },
    };

    return new Promise((resolve, reject) => {
        const req = https.request(options, (res) => {
            let data = '';

            res.on('data', (chunk) => {
                data += chunk;
            });

            res.on('end', () => {
                resolve(JSON.parse(data));
            });
        });

        req.on('error', (error) => {
            reject(error);
        });

        req.end();
    });
}

File: src/requests.py
import http.client
import json
from urllib.parse import quote
import sqlite3

# Fonction pour effectuer la requête GET playlist_id avec le token
def search_playlist_id(token, playlist_name):
    conn = http.client.HTTPSConnection("api.spotify.com")
    headers = {"Authorization": f"Bearer {token}"}
   
    # Encode le paramètre de requête "q"
    encoded_playslist_name = quote(playlist_name, safe='')

    params = f"q={encoded_playslist_name}&type=playlist&limit=1"

    conn.request("GET", f"/v1/search?{params}", headers=headers)
    res = conn.getresponse()
    data = res.read()
    return json.loads(data.decode("utf-8"))

# Fonction pour effectuer la requête GET track avec le token
def search_playlist(token, playlist_id):
    conn = http.client.HTTPSConnection("api.spotify.com")
    headers = {"Authorization": f"Bearer {token}"}
    conn.request("GET", f"/v1/playlists/{playlist_id}", headers=headers)
    res = conn.getresponse()
    data = res.read()
    return json.loads(data.decode("utf-8"))

# Fonction pour effectuer la requête GET track avec le token
def search_track(token, track_name):
    conn = http.client.HTTPSConnection("api.spotify.com")
    headers = {"Authorization": f"Bearer {token}"}
   
    # Encode le paramètre de requête "q"
    encoded_track_name = quote(track_name, safe='')

    params = f"q={encoded_track_name}&type=track&limit=1"

    conn.request("GET", f"/v1/search?{params}", headers=headers)
    res = conn.getresponse()
    data = res.read()
    return json.loads(data.decode("utf-8"))

